src/core/styles/components/tiem-picker/_time-picker-theme.scss

```diff
@@ -0,0 +1,202 @@
+////
+/// @group themes
+/// @access public
+/// @author <a href="https://github.com/desig9stein" target="_blank">Marin Popov</a>
+////
+
+/// If only background color is specified, text/icon color
+/// will be assigned automatically to a contrasting color.
+/// Does ___not___ apply for disabled state colors.
+///
+/// @param {Color} $text-color [null] - The text color of a open time picker.
+/// @param {Color} $hover-text-color [null] - The hover text color of a open time picker.
+/// @param {Color} $selected-text-color [null] -  The text color of a selected item in time picker.
+/// @param {Color} $activeList-item-background [null] - The background color for current item in focused column inside the time picker.
+/// @param {Color} $igx-time-picker-header-background [null] - The header background color of a time picker.
+/// @param {Color} $igx-time-picker-header-hour-text-color [null] - The header hour text color of a time picker.
+/// @param {Color} $igx-time-picker-header-ampm-text-color [null] - The header AM/PM text color of a time picker.
+///
+/// @requires text-contrast
+///
+/// @example scss Change the background and text colors in  time picker
+///   $my-time-picker-theme: igx-time-picker-theme(
+///     $text-color: white,
+///     $background-color: black
+///   );
+///   // Pass the theme to the igx-time-picker component mixin
+///   @include igx-time-picker($my-button-theme);
+
+@function igx-time-picker-theme(
+    $text-color: null,
+    $hover-text-color: null,
+    $selected-text-color: null,
+    $activeList-item-background: null,
+    $igx-time-picker-header-background: null,
+    $igx-time-picker-header-hour-text-color: null,
+    $igx-time-picker-header-ampm-text-color: null,
```

Comment:
The $igx-time-picker property prefixes are unnecessary, making the variable names too long and inconsistent  with the other property names. Please refactor without this prefix. Also instead of naming the time period parameter "ampm" something like `header-time-period-color` would be more descriptive.